maat-court-data-api/src/test/java/gov/uk/courtdata/assessment/validator/UpdateAppDateCompletedValidatorTest.java

```diff
@@ -0,0 +1,70 @@
+package gov.uk.courtdata.assessment.validator;
+
+import gov.uk.courtdata.exception.ValidationException;
+import gov.uk.courtdata.model.assessment.UpdateAppDateCompleted;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+
+@ExtendWith(MockitoExtension.class)
+class UpdateAppDateCompletedValidatorTest {
+
+    @InjectMocks
+    private UpdateAppDateCompletedValidator updateAppDateCompletedValidator;
+
+
+    @Test
+    public void testUpdateAppDateCompletedValidator_whenRepIdIDIsDefine_thenValidationPasses() {
+        Optional<Void> result = updateAppDateCompletedValidator.validate(getUpdateAppDateCompletedWithRepId(1000));
+        assertThat(result).isEqualTo(Optional.empty());
+    }
+
+    @Test
+    public void testUpdateAppDateCompletedValidator_whenRepIdIDIsNull_thenValidationPasses() {
+        ValidationException validationException = Assertions.assertThrows(ValidationException.class,
+                () -> updateAppDateCompletedValidator.validate(getUpdateAppDateCompletedWithRepId(null)));
+        assertThat(validationException.getMessage()).isEqualTo("Rep Id is missing from request and is required");
+    }
+
+    @Test
+    public void testUpdateAppDateCompletedValidator_whenAppDateCompletedIsDefine_thenValidationPasses() {
+        Optional<Void> result = updateAppDateCompletedValidator.validate(getUpdateAppDateCompleted(LocalDate.now()));
+        assertThat(result).isEqualTo(Optional.empty());
+    }
+
+    @Test
+    public void testUpdateAppDateCompletedValidator_whenAppDateCompletedIsNull_thenValidationPasses() {
```

Comment:
The name of this test is a little misleading.

Reply:
Fixed now